### ISSUE TITLE
Crashing on pressing 3D map

### ISSUE THREAD

**Issue by edgatas** [EDDiscovery/EDDiscovery#343]
The program crashes while pressing 3D map button. Attaching both log files.
Windows 10, 64 bit system.

[Logs.zip](https://github.com/EDDiscovery/EDDiscovery/files/423600/Logs.zip)

**Comment by klightspeed** [comment]
@robbyxp1 
VisitedSystemsClass.cs:304
```
            Dictionary<long, ISystem> systems = distlist.Values.ToDictionary(s => s.id);
```

If there are two identical distances, this will throw the DuplicateKeyException.

I will submit a fix to this.

**Comment by edgatas** [comment]
No, I'm not doing anything. However, I might have found a reason for that. I found out that my netLog data was corrupted for a few days and it messed my map on both EDD and EDSM. I will try to exclude those and fix those netLogs someday.

**Comment by edgatas** [comment]
Hello again. 
Version 4.2.1 The same problem with pressing the map button, but now it shows exceptions very nicely in the program itself. Very nice job on that.
From what I understand it could be a problem from my end.
Attaching the Trace file.
[Trace_20160820114203.zip](https://github.com/EDDiscovery/EDDiscovery/files/428232/Trace_20160820114203.zip)

**Comment by klightspeed** [comment]
That looks like something went wrong in the OpenGL library.

The initial `BadImageFormatException` seems to suggest that you have a 32-bit `libEGL.dll` somewhere in your `PATH`.

**Comment by klightspeed** [comment]
4.2.2 (to be released in the next day or so) should contain a fix for this (#347).  If it continues to crash with 4.2.2, please let us know.

**Comment by alexcraig** [comment]
I'm having a similar issue with 4.3.3. When I click the "3D map" button, the map panel doesn't appear, and the application stops responding indefinitely. The issue just started happening the last time I closed and opened the application, and this release was working fine previously. I tried removing the "Data" folder and letting EDD regenerate it, but the issue persists.

Windows 7, 64 bit system. Attached is the trace file. I don't know where to find the other log file the original reporter attached.

**Comment by klightspeed** [comment]
Nothing seems to be logged on the hang.  EDDiscovery does need to load a lot of data while opening the 3D map, and we are looking at performance issues on mechanical HDDs.

The last event logged is from the end of the systems sync - perhaps you tried to open the 3D map while EDDiscovery was performing a full sync, and this resulted in a deadlock.

Was there any disk activity while it was hung?

**Comment by alexcraig** [comment]
I get disk activity for the first ~5 - 10 seconds of the hang, after which there is no disk activity. Both ED and EDDiscovery are installed on an SSD.

I also tried a completely fresh install of EDDiscovery, and the same thing happens. I opened the program once, let it fetch the EDSM data, set my log location, and closed it. I then started it again, let it parse my logs, and tried to open the 3D map, at which point it hangs.
